src/main/scala/ch/epfl/bluebrain/nexus/admin/index/DistributedDataIndex.scala

```diff
@@ -0,0 +1,188 @@
+
+  private val labelToProjectsKey      = LWWMapKey[String, ResourceF[Project]]("labels_to_projects")
+  private val labelToOrganizationsKey = LWWMapKey[String, ResourceF[Organization]]("labels_to_orgs")
+  private val organizationsListKey    = LWWRegisterKey[RevisionedValue[SortedSet[ResourceF[Organization]]]]("orgs_list")
+  private val projectsListKey         = LWWRegisterKey[RevisionedValue[SortedSet[ResourceF[Project]]]]("projects_list")
+
+  private def projectsOrgListKey(org: UUID): LWWRegisterKey[RevisionedValue[SortedSet[ResourceF[Project]]]] =
+    LWWRegisterKey[RevisionedValue[SortedSet[ResourceF[Project]]]](s"${org.toString}_projects_list")
+
+  private implicit val orgOrdering: Ordering[ResourceF[Organization]] = Ordering.by { org: ResourceF[Organization] =>
+    org.value.label
+  }
+
+  private implicit val projectOrdering: Ordering[ResourceF[Project]] = Ordering.by { proj: ResourceF[Project] =>
+    s"${proj.value.organization}/${proj.value.label}"
+  }
+
+  override def updateOrganization(organization: ResourceF[Organization]): F[Boolean] =
+    for {
+      _ <- updateUuidToResource(labelToOrganizationsKey, organization.value.label, organization)
+      _ <- updateList(organizationsListKey, organization)
+    } yield true
+
+  override def updateProject(project: ResourceF[Project]): F[Boolean] = {
+
+    def unwrapUuid(orgOpt: Option[ResourceF[Organization]]) = orgOpt match {
+      case None      => F.raiseError(UnexpectedState(project.uuid.toString))
+      case Some(org) => F.pure(org.uuid)
+    }
+
+    for {
+      _    <- updateUuidToResource(labelToProjectsKey, project.value.fullLabel, project)
+      _    <- updateList(projectsListKey, project)
+      org  <- getOrganization(project.value.organization)
+      uuid <- unwrapUuid(org)
+      _    <- updateList(projectsOrgListKey(uuid), project)
+    } yield true
+  }
+
+  override def getOrganization(label: String): F[Option[ResourceF[Organization]]] =
+    fetchFromMap(labelToOrganizationsKey, label)
+
+  override def getProject(organization: String, project: String): F[Option[ResourceF[Project]]] =
+    fetchFromMap(labelToProjectsKey, s"$organization/$project")
+
+  override def listOrganizations(pagination: Pagination): F[List[ResourceF[Organization]]] =
+    fetchList(organizationsListKey, pagination)
+
+  override def listProjects(pagination: Pagination): F[List[ResourceF[Project]]] =
+    fetchList(projectsListKey, pagination)
+
+  override def listProjects(organizationLabel: String, pagination: Pagination): F[List[ResourceF[Project]]] = {
+    getOrganization(organizationLabel).flatMap {
+      case None => F.pure(List.empty)
+      case Some(org) =>
+        fetchList(projectsOrgListKey(org.uuid), pagination)
+    }
+  }
+
+  private def fetchList[A](dDataKey: LWWRegisterKey[RevisionedValue[SortedSet[ResourceF[A]]]],
+                           pagination: Pagination): F[List[ResourceF[A]]] = {
+    val get = Get(dDataKey, ReadMajority(consistencyTimeout), None)
```

Comment:
I think we should use ReadLocal and WriteAll consistencies; we expect little writes and a lot of reads.